ads/google/doubleclick.js

```diff
@@ -87,6 +135,79 @@ function doubleClickWithGpt(global, data, gladeExperiment) {
   container.style.right = '';
   container.style.transform = 'translate(-50%, -50%)';
 
+  // Get multi-size ad request data, if any, and validate it in the following
+  // ways:
+  // 1) Ensure that the data string is a comma-separated list of sizes of the
+  //    form WxH;
+  // 2) that each secondary dimension is strictly less than its primary
+  //    dimension counterpart;
+  // 3) and, if data-mutli-size-validation is not set to false, that each
+  //    secondary dimension is at least 2/3rds of its primary dimension
+  //    counterpart.
+  const multiSizeDataStr = data.multiSize || null;
+  if (multiSizeDataStr) {
+    const sizesStr = multiSizeDataStr.split(',');
+    sizesStr.forEach(sizeStr => {
+
+      const size = sizeStr.split('x');
+
+      // Make sure that each size is specified in the form val1xval2.
+      if (size.length != 2) {
+        user().error('AMP-AD', `Invalid multi-size data format '${sizeStr}'.`);
+        return;
+      }
+
+      const widthStr = size[0];
+      const heightStr = size[1];
+
+      // Make sure that both dimensions given are numbers.
+      if (!validateDimensions(widthStr, heightStr,
+            w => isNaN(Number(w)),
+            h => isNaN(Number(h)),
+            ({badDim, badVal}) =>
```

Comment:
It's just for array deconstruction, since it includes giant Symbol polyfills.

It's called assignment deconstruction depending on the context.